internal/blockchain/indexers/indexsubscriber.go

```diff
@@ -409,10 +391,30 @@ func (s *IndexSubscriber) handleIndexUpdates(ctx context.Context) {
 //
 // This should be run as a goroutine.
 func (s *IndexSubscriber) Run(ctx context.Context) {
-	s.wg.Add(2)
-	go s.handleIndexUpdates(ctx)
-	go s.handleSyncSubscribers(ctx)
-	s.wg.Wait()
-
-	log.Infof("Index subscriber shutting down")
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
+		s.handleIndexUpdates(ctx)
+		wg.Done()
+	}()
+	go func() {
+		s.handleSyncSubscribers(ctx)
+		wg.Done()
+	}()
```

Comment:
:+1: This pattern give a guarantee at a glance.  The verbosity here is well worth it.